compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partitionning.rs

```diff
@@ -169,7 +172,15 @@ fn resolve_by_levelled_block(
             1 => get_singleton_value(&constraints.forced),
             _ => {
                 let forced = constraints.forced;
-                if forced.contains(&default_partition) {
+                // in case of conflict, prioritize the external partition, then the default partition
+                let externals: Vec<_> = forced
+                    .iter()
+                    .filter(|p| p_cut.is_external_partition(p))
```

Comment:
I think you can replace the lambda by `Type::method`, but to verify.

Reply:
then how to pass the p_cut? the method need a self and a partition

Reply:
oh right I misread.